src/adapter.test.ts

```diff
@@ -215,3 +216,131 @@ Determine what to do:
   const res2 = await toPromise(actor2);
   expect(res2?.tool).toEqual('rateJoke');
 });
+
+test('fromEvent - ', async () => {
+  const openAi = new OpenAI({
+    apiKey: process.env.OPENAI_API_KEY,
+  });
+
+  const adapter = createOpenAIAdapter(openAi, {
+    model: 'gpt-3.5-turbo',
+  });
+
+  const schemas = createSchemas({
+    context: {
+      storedNumber: {
+        type: 'number',
+        description: 'The stored number',
+      },
+      label: {
+        type: 'string',
+        description: 'The label of the number',
+      },
+    },
+    events: {
+      'number.evaluate': {
+        description:
+          'Evaluate whether the number passed as input is even or odd.',
+        properties: {
+          number: {
+            type: 'number',
+            description: 'The input number to evaluate',
+          },
+        },
+      },
+      proclaimEvenNumber: {
+        description: 'Show the even number',
+        properties: {},
+      },
+      proclaimOddNumber: {
+        description: 'Show the odd number',
+        properties: {},
+      },
+    },
+  });
+
+  const machine = setup({
+    schemas,
+    types: schemas.types,
+    actions: {
+      storeNumber: assign({
+        storedNumber: (_, params: { number: number }) => params.number,
+      }),
+      labelNumberAsEven: assign({ label: 'Even' }),
+      labelNumberAsOdd: assign({ label: 'Odd' }),
+    },
+    actors: {
+      evaluate: adapter.fromEvent(
+        (input: number) =>
+          `Decide what to do based on the given input, which may be an even number or an odd number: ${input}`
+      ),
+    },
+  }).createMachine({
+    id: 'numberDisplayer',
+    initial: 'idle',
+    context: {
+      storedNumber: 0,
+      label: '',
+    },
+    states: {
+      idle: {
+        on: {
+          'number.evaluate': {
+            target: 'evaluate',
+            actions: [
+              {
+                type: 'storeNumber',
+                params: ({ event }) => ({
+                  number: event.number,
+                }),
+              },
+            ],
+          },
+        },
+      },
+      evaluate: {
+        invoke: {
+          src: 'evaluate',
+          input: ({ event }) => {
+            if ('number' in event) {
+              return event.number;
+            }
+            throw new Error('Invalid input');
```

Comment:
Is it expected to do the type narrowing here for the one event I'm interested in, the one with `event.number`? Also, I had to throw an error outside of the conditional to pass TS.